app/src/main/java/com/keylesspalace/tusky/components/occurrence/OccurrenceEntity.kt

```diff
@@ -0,0 +1,73 @@
+/* Copyright Tusky Contributors
+ *
+ * This file is a part of Tusky.
+ *
+ * This program is free software; you can redistribute it and/or modify it under the terms of the
+ * GNU General Public License as published by the Free Software Foundation; either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * Tusky is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even
+ * the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General
+ * Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with Tusky; if not,
+ * see <http://www.gnu.org/licenses>. */
+
+package com.keylesspalace.tusky.components.occurrence
+
+import androidx.room.Entity
+import androidx.room.PrimaryKey
+import androidx.room.TypeConverters
+import com.keylesspalace.tusky.db.Converters
+import java.util.*
+
+@Entity
+@TypeConverters(Converters::class)
+data class OccurrenceEntity(
+    @PrimaryKey(autoGenerate = true) val id: Long = 0,
+    val accountId: Long? = null,
+    val type: Type,
+    val what: String,
+    val startedAt: Date, // TODO or use LocalDateTime (or Long)?
+    val finishedAt: Date? = null,
+    val code: Int? = null,
+    val callTrace: String,
+) {
+    companion object {
+        fun reduceTrace(stackTrace: Array<StackTraceElement>): String {
+            // TODO conditions/transforms here are a bit arbitrary...
+            // TODO we could maybe record more (full stack?) in the db and only reduce it for display?
```

Comment:
These two are the most interesting / to be discussed todos.

(What exactly to record and what to show.)